Add an http -> https redirect frontend and service.

brendandburns opened the pull request:
Wire it up to the ingress.

@eparis @rmmh @fejta

fejta commented on mungegithub/submit-queue/deployment/nginx-redirect/nginx.conf:
```diff
@@ -0,0 +1,41 @@
+    listen 80;
+    server_name contrib.submit-queue.k8s.io;
+    location / {
+        if ($http_user_agent ~ (GoogleHC) ) {
```
What is GoogleHC?

brendandburns replied:
That's the user-agent that the Google HTTP LoadBalancer sends for it's health checks.

fejta commented:
/lgtm

fejta commented:
Any chance this regressed: https://storage.googleapis.com/kubernetes-test-history/k8s-queue-health.svg

fejta commented:
(not sure if that's still updating or why the timestamp on the bottom left is red)

eparis commented on mungegithub/submit-queue/deployment/nginx-redirect/nginx-deploy.yaml:
```diff
@@ -0,0 +1,23 @@
+      volumes:
+      - name: conf
+        configMap:
+          name: nginx-https-redirect
```
what is this configMap?

brendandburns replied:
It contains the contents of the `nginx.conf` file in this PR. Would you like me to include the config map as well?

I created it with:

```
kubectl create cm nginx-https-redirect --from-file=nginx.conf=nginx.conf
```

(I could add a README too, I suppose)

brendandburns commented:
@fejta I don't know how that graph is created, do you have a pointer to the code?

The 'sample' time certainly corresponds to when I pushed the config.

Would you like me to rollback?  (or I'm happy to fix up that picture...)

Thanks
--brendan

fejta commented:
Code is here: https://github.com/kubernetes/test-infra/blob/master/queue-health/poll/poller.py

Deployment is here: https://github.com/kubernetes/test-infra/blob/master/queue-health/queue-health-prod.yaml#L27

I suspect maybe the code either does not redirect or else rejects the cert?

brendandburns commented:
I added a README.md to explain the ConfigMap

I also fixed `poller.py` to handle both `https` and redirects (should they be needed in the future).

I don't think I have permissions to redeploy the poller, so either someone needs to merge/rebuild, or I need permissions to push a new version of that container.

fejta commented:
Thanks! Still /lgtm

We should probably migrate this to gcr.io/k8s-testimages. I can give you access to that if you want to push the image there.

And if you merge it I'll update the deployment.

brendandburns commented:
I pushed gcr.io/k8s-testimages/queue-health-poll:v20170407 with these fixes. I updated the image in the checked-in poller config as well.

Checking the logs for the running poller, things seem to be restored.

brendandburns commented:
@k8s-bot bazel test this
